homeassistant/components/fritzbox/light.py

```diff
@@ -0,0 +1,182 @@
+        """Return if lightbulb is available."""
+        return self.device.present  # type: ignore [no-any-return]
+
+    @property
+    def is_on(self) -> bool:
+        """If the light is currently on or off."""
+        return self.device.state  # type: ignore [no-any-return]
+
+    @property
+    def brightness(self) -> int:
+        """Return the current Brightness."""
+        return self.device.level  # type: ignore [no-any-return]
+
+    @property
+    def hs_color(self) -> tuple[float, float] | None:
+        """Return the hs color value."""
+        # Don't return hue and saturation unless in color mode
+        if self.device.color_mode != "1":
+            return None
+
+        hue = self.device.hue
+        saturation = self.device.saturation
+
+        return (hue, float(saturation) * 100.0 / 255.0)
+
+    @property
+    def color_temp(self) -> int | None:
+        """Return the CT color value."""
+        # Don't return color temperature unless in color temperature mode
+        if self.device.color_mode != "4":
+            return None
+
+        kelvin = self.device.color_temp
+        return color.color_temperature_kelvin_to_mired(kelvin)
+
+    @property
+    def supported_color_modes(self) -> set:
+        """Flag supported color modes."""
+        return SUPPORTED_COLOR_MODES
+
+    async def async_turn_on(self, **kwargs: Any) -> None:
+        """Turn the light on."""
+        if kwargs.get(ATTR_BRIGHTNESS) is not None:
+            level = kwargs[ATTR_BRIGHTNESS]
+            await self.hass.async_add_executor_job(self.device.set_level, level)
+        if kwargs.get(ATTR_HS_COLOR) is not None:
+            hass_hue = int(kwargs[ATTR_HS_COLOR][0])
+            hass_saturation = round(kwargs[ATTR_HS_COLOR][1] * 255.0 / 100.0)
+            # find supported hs values closest to what user selected
+            hue = min(self._supported_hs.keys(), key=lambda x: abs(x - hass_hue))
+            saturation = min(
+                self._supported_hs[hue], key=lambda x: abs(x - hass_saturation)
+            )
+            await self.hass.async_add_executor_job(
+                self.device.set_color, (hue, saturation)
+            )
+
+        if kwargs.get(ATTR_COLOR_TEMP) is not None:
+            kelvin = color.color_temperature_kelvin_to_mired(kwargs[ATTR_COLOR_TEMP])
+            await self.hass.async_add_executor_job(self.device.set_color_temp, kelvin)
+
+        await self.hass.async_add_executor_job(self.device.set_state_on)
+        await self.coordinator.async_refresh()
+
+    async def async_turn_off(self, **kwargs: Any) -> None:
+        """Turn the light off."""
+        await self.hass.async_add_executor_job(self.device.set_state_off)
+        await self.coordinator.async_refresh()
+
+    @property
+    def extra_state_attributes(self) -> LightExtraAttributes:
+        """Return the state attributes of the device."""
+        attrs: LightExtraAttributes = {
+            ATTR_BRIGHTNESS: self.brightness,
```

Comment:
These are already attributes/properties of the light itself and should not be set "extra".

Reply:
Makes sense 🙈